torch_points3d/datasets/base_dataset.py

```diff
@@ -56,11 +59,57 @@ def batch_size(self) -> int:
     def collate_fn(self) -> Optional[Callable]:
         return torch_geometric.data.batch.Batch.from_data_list
 
+    @staticmethod
+    def get_num_samples(batch, conv_type):
+        is_dense = ConvolutionFormatFactory.check_is_dense_format(conv_type)
+        if is_dense:
+            return batch.pos.shape[0]
+        else:
+            return (batch.batch.max() + 1).item()
+
+    @staticmethod
+    def _collate_fn(batch: Data, collate_fn: Callable, pre_collate_transform: Optional[Callable] = None):
+        if pre_collate_transform:
+            batch = pre_collate_transform(batch)
+        return collate_fn(batch)
+
+    @staticmethod
+    def _get_collate_function(conv_type: str, is_multiscale: bool, pre_collate_transform: Optional[Callable] = None):
+        is_dense: bool = ConvolutionFormatFactory.check_is_dense_format(conv_type)
+        if is_multiscale:
+            if conv_type.lower() == ConvolutionFormat.PARTIAL_DENSE.value.lower():
```

Comment:
I wonder, can we do validation on "conv_type" so that the value is always matching the enum? Maybe once we do static checking with hydra it will do  that for us

Reply:
It would be great to perform static type checking on conv_type. But here we are not doing type checking.